lib/from_openstudio/material/opaque.rb

```diff
@@ -0,0 +1,59 @@
+# *******************************************************************************
+# Honeybee OpenStudio Gem, Copyright (c) 2020, Alliance for Sustainable
+# Energy, LLC, Ladybug Tools LLC and other contributors. All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+#
+# (1) Redistributions of source code must retain the above copyright notice,
+# this list of conditions and the following disclaimer.
+#
+# (2) Redistributions in binary form must reproduce the above copyright notice,
+# this list of conditions and the following disclaimer in the documentation
+# and/or other materials provided with the distribution.
+#
+# (3) Neither the name of the copyright holder nor the names of any contributors
+# may be used to endorse or promote products derived from this software without
+# specific prior written permission from the respective party.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDER(S) AND ANY CONTRIBUTORS
+# "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+# THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER(S), ANY CONTRIBUTORS, THE
+# UNITED STATES GOVERNMENT, OR THE UNITED STATES DEPARTMENT OF ENERGY, NOR ANY OF
+# THEIR EMPLOYEES, BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+# EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT
+# OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+# STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+# OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+# *******************************************************************************
+
+require 'honeybee/material/opaque'
+require 'to_openstudio/model_object'
+
+module Honeybee
+  class EnergyMaterial < ModelObject
+
+    def self.from_material(material)
+        # create an empty hash
+        hash = {}
+        hash[:type] = 'EnergyMaterial'
+        # set hash values from OpenStudio Object
+        hash[:identifier] = clean_identifier(material.nameString)
+        hash[:display_name] = material.nameString
+        hash[:user_data] = {handle: material.handle.to_s}
+        hash[:thickness] = material.thickness
+        hash[:conductivity] = material.conductivity
+        hash[:density] = material.density
+        hash[:specific_heat] = material.specificHeat
+        hash[:roughness] = material.roughness
+        hash[:thermal_absorptance] = material.thermalAbsorptance
+        hash[:solar_absorptance] = material.solarAbsorptance
+        hash[:visible_absorptance] = material.visibleAbsorptance
```

Comment:
I can see that the OpenStudio SDK returns "boost optional" objects for the three variables:
![image](https://user-images.githubusercontent.com/5567574/112914207-6313cb80-90c9-11eb-81ea-5dc0d040d919.png)

And I would bet that "boost optional" objects are not JSON-serializable. So, you'll have to go through the properties of all the objects that you are writing translators for here and check whether they return "boost optional" objects.

If they do, then you need to see if they are empty before you write them into the hash. In almost all cases, the Honeybee default value is the same as the EnergyPlus default value so I don't think you need to worry about what the default should be written into the honeybee hash when the OSM has no value for these optional fields. For the few cases where I know the Honeybee default is different from the E+ one, I will call it out with a review comment.

Reply:
Ah, ok. I just forgot that we are using absorptances in Honeybee-schema instead of reflectances. I see that you're right that OpenStudio SDK returns doubles for all of those relfectances so your code should work.